server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServer.java

```diff
@@ -75,14 +78,20 @@ public GuiceJamesServer overrideWith(List<Module> overrides) {
         return new GuiceJamesServer(isStartedProbe, Modules.override(module).with(overrides));
     }
 
-    public void start() throws Exception {
-        Injector injector = Guice.createInjector(module);
-        preDestroy = injector.getInstance(Key.get(new TypeLiteral<Stager<PreDestroy>>() {}));
-        injector.getInstance(StartUpChecksPerformer.class)
-            .performCheck();
-        injector.getInstance(InitializationOperations.class).initModules();
-        guiceProbeProvider = injector.getInstance(GuiceProbeProvider.class);
-        isStartedProbe.notifyStarted();
+    public void start() {
+        try {
+            Injector injector = Guice.createInjector(module);
+            preDestroy = injector.getInstance(Key.get(new TypeLiteral<Stager<PreDestroy>>() {
+            }));
+            injector.getInstance(StartUpChecksPerformer.class)
+                .performCheck();
+            injector.getInstance(InitializationOperations.class).initModules();
+            guiceProbeProvider = injector.getInstance(GuiceProbeProvider.class);
+            isStartedProbe.notifyStarted();
+        } catch (Exception e) {
+            LOGGER.error("Fatal error while starting James", e);
+            stop();
```

Comment:
aren't you going to generate even more unrelated exception by trying to stop the server? What does `stop` do that terminating the problem doesn't?